src/main/java/org/spongepowered/common/SpongeServer.java

```diff
@@ -47,6 +49,10 @@ public interface SpongeServer extends SpongeEngine, Server {
 
     UsernameCache getUsernameCache();
 
+    @Nullable Integer getBlockDestructionId(BlockPos pos);
```

Comment:
Shouldn't that be on the world?

Reply:
Minecraft renderer makes no difference between worlds so making this per-world may lead to some weird interferences. For example:
1. Player in World №1. We send progress to block №1.
2. Player travels to World №2. We send progress to block №2.
3. Player travels back to World №1. And 2 results can happen:
3.1. If packet in (2) had the same id as in (1) (pretty random thing), progress on block №1 will be reset and player will not see it.
3.2. If packets had different ids, progress on block №1 will remain and player will see it.

This is some kind of indeterminacy and generally I don't like it. But maybe we don't even want to somehow support this described use case so I'm not sure.